### DISCUSSION TITLE
DPD issue on after upgrading to strongswan 5.9.6

### DISCUSSION THREAD

**Discussion by zfortna** [strongswan/strongswan#1659] [General]
[config.txt](https://github.com/strongswan/strongswan/files/11260793/config.txt)

After upgrading from 5.9.4 to 5.9.6 we are now seeing a unexpected behavior when the SeGW on the other end of the tunnel performs a failover.  When the failover occurs on the SeGW that strongswan is connected to, the traffic is directed to the secondary SeGW for 2-3 seconds and then killed.  On the strongswan side, it detects the failure as a dead peer and falls into DPD detection until strongswan is restarted.  

Could this be related to the feature that was added in 5.9.6:
Actively initiating duplicate CHILD_SAs within the same IKE_SA is now largely prevented. This can happen if trap policies are installed and an IKE_SA with its CHILD_SAs is reestablished (e.g. with break-before-make reauthentication or dpd_action=restart). This does not prevent duplicates if they are initiated by the two peers concurrently.

If it is related, is there a way to prevent this behavior from occurring? Duplicheck is already disabled as this configuration requires reauth in make before break configuration.  

I'd prefer to make the necessary adjustments (if any are available) instead of downgrading to strongswan 5.9.4 if it is possible.  

Configuration for the site is attached.

**Comment by tobiasbrunner** [comment]
> when the SeGW on the other end of the tunnel performs a failover.

What does that mean exactly?

> the traffic is directed to the secondary SeGW for 2-3 seconds and then killed.

What exactly does that mean?

> detects the failure as a dead peer and falls into DPD detection until strongswan is restarted.

Detects what exactly and how? Please provide logs etc.

> Duplicheck is already disabled as this configuration requires reauth in make before break configuration.

Do you mean the _duplicheck_ plugin (which really should never be enabled)? Or uniqueness checks (which should work fine with make-before-break reauth)?

**Reply by zfortna** [reply]
The SeGW that strongswan connects to is in a redundant active/passive configuration.  The issue I am seeing is when the active node of the SeGW fails and the SeGW fails over to the passive node.  When this failover occurs, the passive node see the traffic that was being sent to the active node for 2-3 seconds before the traffic is terminated.  

Logs showing what strongswan sees when the failover occurs are attached.  
[dpdlogs.txt](https://github.com/strongswan/strongswan/files/11261507/dpdlogs.txt)

I was referring to the charon duplicheck plugin is set to load=no.  To my knowledge no other changes are made in regards to uniqueness checks.

**Reply by tobiasbrunner** [reply]
> The SeGW that strongswan connects to is in a redundant active/passive configuration. The issue I am seeing is when the active node of the SeGW fails and the SeGW fails over to the passive node. When this failover occurs, the passive node see the traffic that was being sent to the active node for 2-3 seconds before the traffic is terminated.

Is that based on strongSwan's HA solution? Or some other kind of mechanism? How is traffic terminated? Just doesn't get through anymore?

> Logs showing what strongswan sees when the failover occurs are attached.

That does not show any retransmits, reauthentication or any errors.

**Reply by zfortna** [reply]
The SeGW being connected to is an appliance based solution from Juniper/Maviner/etc configured in active/passive mode.  The SeGW is the same product/version that was previously tested on 5.9.4 without an issue.  If a ping is run from server side of the tunnel to the inner IP on the strongswan side of the tunnel, the ping stops in the 2-3 second window after failover occurs.  Also from the SeGW side, it can see the stream of traffic sent from the strongswan client stop 2-3 seconds after failover.

**Reply by tobiasbrunner** [reply]
I don't see anything going wrong (or happening at all really) on the strongSwan side. So since it's unaware of any changes, I'd assume it's more of on issue on the responder side.
